haskellPackages.glicko: unbreak

Prillan opened the pull request:
I'm the author of the package and fixed the dependencies upstream, now builds correctly and can be unmarked as broken.

Tested with current unstable (4eccd6f73).
```
nix-shell -p 'haskellPackages.ghcWithPackages (pkgs: [pkgs.glicko])' -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/4eccd6f731627ba5ad9915bcf600c9329a34ca78.tar.gz
```

iblech commented:
Can confirm that glicko builds just fine, thank you for your work :-)

cdepillabout reviewed (changes requested):
@Prillan Can you rebase this on the `haskell-updates` branch (and make sure to change the base branch here on GitHub as well)?

After you make this change, I'll merge in this PR.

-------------------------------

We generally try to make most Haskell-related changes to the `haskell-updates` branch.  That branch gets merged into `master` about once a week.  Here's the next PR that will get merged into master, in case you want to follow progress: https://github.com/NixOS/nixpkgs/pull/109474

Prillan committed:
haskellPackages.glicko: unbreak

I'm the author of the package and fixed the dependencies upstream,
now builds correctly and can be unmarked as broken.

Prillan commented:
> @Prillan Can you rebase this on the `haskell-updates` branch (and make sure to change the base branch here on GitHub as well)?
> 
> After you make this change, I'll merge in this PR.

Done!

> We generally try to make most Haskell-related changes to the `haskell-updates` branch. That branch gets merged into `master` about once a week. Here's the next PR that will get merged into master, in case you want to follow progress: #109474

Alright, that makes sense, thanks!

cdepillabout commented:
@Prillan Thanks for unbreaking this.

@iblech and thanks for reviewing.
